src/app/functions/server/updateDataBrokerScanProfile.test.ts

```diff
@@ -0,0 +1,166 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { it, expect, jest } from "@jest/globals";
+import { OnerepProfileRow, UpdateableProfileDetails } from "knex/types/tables";
+
+const currentProfileDetails: OnerepProfileRow = {
+  id: 1234,
+  onerep_profile_id: 5678,
+  first_name: "First01",
+  last_name: "last01",
+  middle_name: "",
+  first_names: [],
+  middle_names: [],
+  last_names: [],
+  city_name: "City01",
+  state_code: "NY",
+  addresses: [{ city: "City01", state: "NY" }],
+  phone_numbers: ["8005553534"],
+  date_of_birth: new Date(),
+  created_at: new Date(),
+  updated_at: new Date(),
+};
+
+const newProfileDetails: UpdateableProfileDetails = {
+  first_name: "First01",
+  last_name: "last01",
+  middle_name: "",
+  first_names: [],
+  middle_names: [],
+  last_names: [],
+  addresses: [{ city: "City01", state: "NY" }],
+  phone_numbers: ["8005553534"],
+};
+
+jest.mock("./onerep", () => ({
+  updateProfile: jest.fn(),
+}));
+
+jest.mock("./refreshStoredScanResults");
+
+jest.mock("./logging", () => {
+  return {
+    logger: {
+      info: jest.fn(),
+    },
+  };
+});
+
+beforeEach(() => {
+  jest.spyOn(console, "info").mockImplementation(() => undefined);
+});
+
+describe("Update broker scan profile", () => {
+  it("updates the profile details", async () => {
+    jest.mock("../../../db/tables/onerep_profiles", () => ({
+      getProfileDetails: jest.fn(() => Promise.resolve(currentProfileDetails)),
+      updateProfileDetails: jest.fn(),
+    }));
+    const updateDataBrokerScanProfile = (
+      await import("./updateDataBrokerScanProfile")
+    ).default;
```

Comment:
Question: why do we need to dynamically import the system-under-test in every test?

Reply:
I’m not certain if the would not be a better way, but without the dynamic imports the [updateProfile mock](https://github.com/mozilla/blurts-server/pull/5660/files#diff-3e4b3503e2c47da6fe68726b4b57fc0ea71b08ccfcb2a344ab90423d75433d6cR38-R40) was not being honored.

Reply:
Dammit, it should just work. It could be that the (non-dynamic) import should just be placed after the mock call, but otherwise... I guess not worth spending much effort on.

I so wish Jest was more reliable.

Reply:
Yeah, I’ve tried that and it _should_ work. After spending some time to workaround it I had to resort to the dynamic import.

Reply:
I'm not sure why this isn't working, I know that Jest's old "hoisting" trick doesn't really work in ES modules (which is why Jest needs `babel` to work properly since it transpiles, AIUI). I would suggest that dynamic imports for mocks are probably in our future one way or another :)